controlplane/api/v1alpha1/rke2controlplane_types.go

```diff
@@ -17,48 +17,270 @@ limitations under the License.
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// Rke2ControlPlaneSpec defines the desired state of Rke2ControlPlane
-type Rke2ControlPlaneSpec struct {
+// RKE2ControlPlaneSpec defines the desired state of RKE2ControlPlane
+type RKE2ControlPlaneSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of Rke2ControlPlane. Edit rke2controlplane_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// ServerConfig specifies configuration for the agent nodes.
+	//+optional
+	ServerConfig RKE2ServerConfig `json:"serverConfig,omitempty"`
+
+	// ManifestsConfigMapReference references a ConfigMap which contains Kubernetes manifests to be deployed automatically on the cluster
+	// Each data entry in the ConfigMap will be will be copied to a folder on the control plane nodes that RKE2 scans and uses to deploy manifests.
+	//+optional
+	ManifestsConfigMapReference corev1.ObjectReference `json:"manifestsConfigMapReference,omitempty"`
 }
 
-// Rke2ControlPlaneStatus defines the observed state of Rke2ControlPlane
-type Rke2ControlPlaneStatus struct {
+type RKE2ServerConfig struct {
+	// RKE2CommonNodeConfig is an inline struct that references common attribtes between agent and server nodes
+	v1alpha1.RKE2CommonNodeConfig `json:",inline"`
+
+	// BindAddress describes the rke2 bind address (default: 0.0.0.0).
+	//+optional
+	BindAddress string `json:"bindAddress,omitempty"`
+
+	// AdvertiseAddress IP address that apiserver uses to advertise to members of the cluster (default: node-external-ip/node-ip).
+	//+optional
+	AdvertiseAddress string `json:"advertiseAddress,omitempty"`
+
+	// TLSSan Add additional hostname or IP as a Subject Alternative Name in the TLS cert.
+	//+optional
+	TLSSan []string `json:"tlsSan,omitempty"`
+
+	// ServiceNodePortRange is the port range to reserve for services with NodePort visibility (default: "30000-32767").
+	//+optional
+	ServiceNodePortRange string `json:"service-node-port-range,omitempty"`
+
+	// ClusterDNS is the cluster IP for CoreDNS service. Should be in your service-cidr range (default: 10.43.0.10).
+	//+optional
+	ClusterDNS string `json:"clusterDNS,omitempty"`
+
+	// ClusterDomain is the cluster domain name (default: "cluster.local").
+	//+optional
+	ClusterDomain string `json:"clusterDomain,omitempty"`
+
+	// DisableComponents lists Kubernetes components and RKE2 plugin components that will be disabled.
+	//+optional
+	DisableComponents DisableComponents `json:"disableComponents,omitempty"`
+
+	// LoadBalancerPort Local port for supervisor client load-balancer. If the supervisor and apiserver are not colocated an additional port 1 less than this port will also be used for the apiserver client load-balancer (default: 6444).
+	//+optional
+	LoadBalancerPort int `json:"lbServerPort,omitempty"`
```

Comment:
json tag name is different, they should be the same except formatting

Reply:
Good catch! I will change that!

Reply:
Done!